unison-cli/src/Unison/Codebase/Editor/HandleInput/Update.hs

```diff
@@ -294,64 +311,126 @@ getSlurpResultForUpdate requestedNames slurpCheckNames = do
     -- Furthermore, #wham is both a dependent of #pingpong (B), and a dependency of #newping, so it too is an implicit
     -- term.
     --
-    -- FIXME: this currently only looks at old components; doesn't search for new cycles.
-    --
     -- Running example:
     --
     --   #pingpong.pong => (<#pingpong.ping + 2>, "pong")
     --   #wham          => (<#pingpong.pong + 3>, "wham")
     implicitTerms :: Map TermReferenceId (Term Symbol Ann, Symbol) <-
       liftIO do
-        -- Running example:
-        --
-        --
-        --   { #pingpong }
-        let oldHashes :: Set Hash
-            oldHashes =
-              foldMap (Set.mapMaybe Reference.toHash) updatedNameToOldRefs
-
-        oldHashes & foldMapM \oldHash -> do
-          -- Running example:
-          --
-          --   [ (<#pingpong.pong + 1>, <Nat>),
-          --     (<#pingpong.ping + 2>, <Nat>)
-          --   ]
-          terms <-
-            Codebase.withConnection codebase \conn ->
-              Sqlite.runTransaction conn (Codebase.unsafeGetTermComponent codebase oldHash)
-          pure $
-            terms
-              -- Running example:
-              --
-              --   [ (#pingpong.ping, (<#pingpong.pong + 1>, <Nat>)),
-              --     (#pingpong.pong, (<#pingpong.ping + 2>, <Nat>))
-              --   ]
-              & Reference.componentFor oldHash
-              & List.foldl'
-                ( \acc (ref, (term, _typ)) ->
-                    let -- (D) After getting the entire component of `oldHash`, which has at least one member being
-                        -- updated, we want to keep only the members that are *not* being updated (i.e. those who have
-                        -- no name that is being updated).
-                        --
-                        -- Running example, first time through (processing #pingpong.ping):
-                        --
-                        --   Set.disjoint { "ping" } { "ping" } is false, so don't add to the map.
-                        --
-                        -- Running example, second time through (processing #pingpong.pong):
-                        --
-                        --   Set.disjoint { "ping" } { "pong" } is true, so add
-                        --   #pingpong.pong => (<#pingpong.ping + 2>, { "pong" })) to the map.
-                        notBeingUpdated = Set.disjoint namesBeingUpdated
-                        nameIsUnconflicted name = Set.size (nameToTermRefs name) == 1
-                        names = termRefToNames ref
-                     in if notBeingUpdated names
-                          then case Foldable.find nameIsUnconflicted names {- (E) -} of
-                            Nothing -> acc
-                            Just name -> Map.insert ref (term, name) acc
-                          else acc
-                )
-                Map.empty
+        Codebase.withConnection codebase \conn ->
```

Comment:
I think you can use `Codebase.runTransaction` here.

Reply:
Yes, thanks, I forgot about that